Pages updates (#2083)
cloudflare/terraform-provider-cloudflare: .changelog/2083.txt

```release-note:enhancement

> resource/cloudflare_pages_project: Updates Pages project resource to support API changes

Reviewer: ```suggestion
resource/cloudflare_pages_project: adds support for `always_use_latest_compatibility_date`, `fail_open`, `service_binding` and `usage_model`
```

> resource/cloudflare_pages_project: Updates Pages project resource to support API changes

Reviewer: ```suggestion
resource/cloudflare_pages_project: adds support for `always_use_latest_compatibility_date`, `fail_open`, `service_binding` and `usage_model`
```
